### PR TITLE
setting a custom resource as sink doc

### PR DESCRIPTION
**Pull request by danielgrist**
Adding documentation on how to set up a Custom Resource in order to be used as a sink.

## Proposed Changes 

- Adds a section [here](https://knative.dev/docs/eventing/sink) to explain the steps necessary to use a custom resource as a sink

### CODE REVIEW

**Comment by danielgrist** [comment]
@googlebot I signed it!

File: docs/eventing/sink/_index.md [hunk 1]
```diff
@@ -44,6 +44,35 @@ spec:
       name: my-kafka-sink
 ```
 
+## Setting up a Custom Resource to be used as a sink
+To allow a Custom Resource to act as a sink for events, there are two things needed: 
+
+#### 1. Make the resource Addressable
+To make a Custom Resource Addresable, it needs to contain a `status.address.url`. More information about [Addressable resources](https://github.com/knative/specs/blob/main/specs/eventing/interfaces.md#addressable).
```
Comment (lionelvillard) [inline docs/eventing/sink/_index.md:51]: nit: Addresable -> Addressable

File: docs/eventing/sink/_index.md [hunk 2]
```diff
@@ -44,6 +44,35 @@ spec:
...
+
+#### 2. Create an addressable-resolver ClusterRole
+An addressable-resolver ClusterRole is needed in order to obtain the necessary RBAC rules for the sink to receive events.
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: kafkasinks-addressable-resolver
```
Comment (lionelvillard) [inline docs/eventing/sink/_index.md:59]: maybe say something about `kafkasinks` being an example here and it should be replaced by the name of your CRD sink?

**Comment by Shashankft9** [comment]
One question, not related to this PR but this just made me wonder - how does eventing get the addressable url from any custom resource? I thought ideally a component like eventing would need to use the client of that custom resource to get the url or any such operations, Is this where knative's duck typing is used?

**Review by lionelvillard — APPROVED** [review]

**Pull request merged**